src/lib.rs

```diff
@@ -246,14 +246,24 @@ impl<Fs: FileSystem + Default> ResolverGeneric<Fs> {
             return Ok(path);
         }
 
-        match specifier.as_bytes()[0] {
+        match Path::new(specifier).components().next() {
             // 3. If X begins with './' or '/' or '../'
-            b'/' | b'\\' => self.require_absolute(cached_path, specifier, ctx),
+            Some(Component::RootDir | Component::Prefix(_)) => {
+                self.require_absolute(cached_path, specifier, ctx)
+            }
             // 3. If X begins with './' or '/' or '../'
-            b'.' => self.require_relative(cached_path, specifier, ctx),
+            Some(Component::CurDir | Component::ParentDir) => {
+                self.require_relative(cached_path, specifier, ctx)
+            }
             // 4. If X begins with '#'
-            b'#' => self.require_hash(cached_path, specifier, ctx),
+            Some(Component::Normal(_)) if specifier.as_bytes()[0] == b'#' => {
+                self.require_hash(cached_path, specifier, ctx)
+            }
             _ => {
+                debug_assert!(Path::new(specifier)
+                    .components()
+                    .next()
+                    .is_some_and(|c| matches!(c, Component::Normal(_))));
```

Comment:
It would be nice to add a message in the assertion that explains the assertion itself